File: src/main/java/net/bms/genera/custom/Faerie.java
package net.bms.genera.custom;

import com.google.gson.Gson;
import com.google.gson.JsonObject;
import net.bms.genera.entities.passive.EntityFaerie;
import net.minecraft.world.World;
import net.minecraftforge.registries.IForgeRegistryEntry;

import java.io.File;
import java.io.FileReader;
import java.io.IOException;

public class Faerie extends IForgeRegistryEntry.Impl<Faerie> {
    private JsonObject jsonObject;

    public Faerie(File file) throws IOException {
        FileReader reader = new FileReader(file);
        Gson jsonParser = new Gson();
        jsonObject = jsonParser.fromJson(reader, JsonObject.class);
    }

    public void spawn(World world, double x, double y, double z) {
        if (!world.isRemote) {
            double maxHealth = jsonObject.get("default_health").getAsDouble();
            String type = jsonObject.get("faerie_type").getAsString();
            float size = jsonObject.get("default_size").getAsFloat();
            int level = jsonObject.get("default_level").getAsInt();
            EntityFaerie faerie = new EntityFaerie(world, maxHealth, type, size, level);
            faerie.setPosition(x, y, z);
            world.spawnEntity(faerie);
        }
    }

    public String getBodyTexture() {
        return jsonObject.get("body_texture").getAsString();
    }

    public String getWingTopTexture() {
        return jsonObject.get("wing_top_texture").getAsString();
    }

    public String getWingBottomTexture() {
        return jsonObject.get("wing_bottom_texture").getAsString();
    }

    public String getName() {
        return jsonObject.get("faerie_type_local").getAsString();
    }
}

File: src/main/kotlin/net/bms/genera/custom/Faerie.kt
package net.bms.genera.custom

import com.google.gson.Gson
import com.google.gson.JsonObject
import net.bms.genera.entity.EntityFaerie
import net.minecraft.world.World
import net.minecraftforge.registries.IForgeRegistryEntry
import java.io.File
import java.io.FileReader
import java.io.IOException

class Faerie @Throws(IOException::class)
constructor(file: File): IForgeRegistryEntry.Impl<Faerie>() {
    private val jsonObject: JsonObject

    val bodyTexture: String
        get() = jsonObject.get("body_texture").asString

    val name: String
        get() = jsonObject.get("faerie_type_local").asString

    val type: String
        get() = jsonObject.get("faerie_type").asString

    init {
        val reader = FileReader(file)
        val jsonParser = Gson()
        jsonObject = jsonParser.fromJson(reader, JsonObject::class.java)
    }

    fun spawn(world: World, x: Double, y: Double, z: Double) {
        if (!world.isRemote) {
            val maxHealth = jsonObject.get("default_health").asDouble
            val type = jsonObject.get("faerie_type").asString
            val size = jsonObject.get("default_size").asFloat
            val level = jsonObject.get("default_level").asInt
            val faerie = EntityFaerie(world, maxHealth, type, size, level)
            faerie.setPosition(x, y, z)
            world.spawnEntity(faerie)
        }
    }
}
